record.go

```diff
@@ -68,15 +70,45 @@ func (r *Record) fromJSONMap(jsonMap map[string]interface{}) {
 			}
 		} else {
 			if v != nil {
-				if _, is := v.(map[string]interface{}); is == false {
+				if obj, is := v.(map[string]interface{}); is == false {
 					r.fields[k] = v
-
+				} else {
+					if r.isLookUp(obj) {
+						if rec, err := RecordFromJSONMap(obj); err == nil {
+							r.lookUps[rec.sobject] = rec
+						}
+					}
 				}
 			}
 		}
 	}
 }
 
+func (r *Record) isLookUp(jsonMap map[string]interface{}) bool {
+	_, has := jsonMap[RecordAttributes]
+	return has
+}
+
+// LookUps returns all of the record's look ups
+func (r *Record) LookUps() []*Record {
+	records := make([]*Record, len(r.lookUps))
+	idx := 0
+	for _, r := range r.lookUps {
+		records[idx] = r
+		idx++
+	}
+	return records
+}
+
+// LookUp returns the look up record
+func (r *Record) LookUp(lookUp string) (*Record, bool) {
+	if len(r.lookUps) == 0 {
+		return nil, false
+	}
+	rec, has := r.lookUps[lookUp]
```

Comment:
Consider just returning here.

Reply:
I don't think you can return the value, ok like this 
```
return r.lookUps[lookUp]
```
for `*Record, bool`

Reply:
Oh dang!  You're right!